core/templates/dev/head/services/assets-backend-api.service.spec.ts

```diff
@@ -108,20 +147,158 @@ describe('Assets Backend API Service', function() {
 
-    var requestUrl = UrlInterpolationService.interpolateUrl(
-      '/assetsdevhandler/exploration/<exploration_id>/assets/image/<filename>',
-      {
-        exploration_id: '0',
-        filename: 'myfile.png'
+    $httpBackend.expect('GET', audioRequestUrl).respond(201, 'audio data');
+    expect(AssetsBackendApiService.isCached('myfile.mp3')).toBe(false);
+    AssetsBackendApiService.loadAudio('0', 'myfile.mp3').then(
+      successHandler, failHandler);
+    $httpBackend.flush();
+    expect(AssetsBackendApiService.isCached('myfile.mp3')).toBe(true);
+
+    expect(successHandler).toHaveBeenCalled();
+    expect(failHandler).not.toHaveBeenCalled();
+    $httpBackend.verifyNoOutstandingExpectation();
+
+    AssetsBackendApiService.loadAudio('0', 'myfile.mp3').then(
+      function(cachedFile) {
+        expect(cachedFile).toEqual(audioFileObjectFactory.createNew(
+          'myfile.mp3',
+          new Blob()
+        ));
       });
+    $httpBackend.verifyNoOutstandingExpectation();
+  });
 
-    $httpBackend.expect('GET', requestUrl).respond(201, 'image data');
-    expect(AssetsBackendApiService.isCached('myfile.png')).toBe(false);
+  it('should handler rejection on fetching file when Blob throws a' +
+    ' TypeErrow and BlobBuilder does not exist', function() {
+    var successHandler = jasmine.createSpy('success');
+    var failHandler = jasmine.createSpy('fail');
 
+    spyOn(window, 'Blob').and.callFake(function() {
+      throw new TypeError();
+    });
+
+    $httpBackend.expect('GET', audioRequestUrl).respond(201, 'Error');
+    expect(AssetsBackendApiService.isCached('myfile.mp3')).toBe(false);
+
+    AssetsBackendApiService.loadAudio('0', 'myfile.mp3').then(
+      successHandler, failHandler);
+    $httpBackend.flush();
+    expect(successHandler).not.toHaveBeenCalled();
+    expect(failHandler).toHaveBeenCalledWith('myfile.mp3');
+    $httpBackend.verifyNoOutstandingExpectation();
+  });
+
+  it('should handler rejection on fetching file when Blob throws a' +
+    ' TypeError and BlobBuilder does not work correctly', function() {
+    var successHandler = jasmine.createSpy('success');
+    var failHandler = jasmine.createSpy('fail');
+
+    spyOn(window, 'Blob').and.callFake(function() {
+      throw new TypeError();
+    });
+
+    window.BlobBuilder = function BlobBuilder() {
+      this.blob = new Object({ blobBuilder: true });
+    };
+
+    $httpBackend.expect('GET', audioRequestUrl).respond(201, 'Error');
+    expect(AssetsBackendApiService.isCached('myfile.mp3')).toBe(false);
+
+    AssetsBackendApiService.loadAudio('0', 'myfile.mp3').then(
+      successHandler, failHandler);
+    $httpBackend.flush();
+    expect(successHandler).not.toHaveBeenCalled();
+    expect(failHandler).toHaveBeenCalledWith('myfile.mp3');
+    $httpBackend.verifyNoOutstandingExpectation();
+  });
+
+  it('should fetch file when Blob throws a TypeError and BlobBuilder' +
+    ' is correctly implemented', function() {
```

Comment:
`is correctly implemented`, it should be `BlobBuilder exist`, right? (I'm not sure, can you please check?)

Reply:
`is correctly implemented` means that `BlobBuilder` has all the methods it should have in order to the code runs fine. `BlobBuild` can exist but it may not have some methods and tests will fail because of it.